File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 09/Go/main.go
package main
import "fmt"

type Imprimivel interface {
    Imprimir()
}

type Relatorio struct {
    Conteudo string
}

func (r Relatorio) Imprimir() {
    fmt.Println("Imprimindo relatório:", r.Conteudo)
}

type Contrato struct {
    Detalhes string
}

func (c Contrato) Imprimir() {
    fmt.Println("Imprimindo contrato:", c.Detalhes)
}

func main() {
    relatorio := Relatorio{Conteudo: "Relatório financeiro"}
    contrato := Contrato{Detalhes: "Contrato de prestação de serviços"}

    relatorio.Imprimir()
    contrato.Imprimir()
}

File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 09/Java/Main.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    private String conteudo;

    public Relatorio(String conteudo) {
        this.conteudo = conteudo;
    }

    @Override
    public void imprimir() {
        System.out.println("Imprimindo relatório: " + conteudo);
    }
}

class Contrato implements Imprimivel {
    private String detalhes;

    public Contrato(String detalhes) {
        this.detalhes = detalhes;
    }

    @Override
    public void imprimir() {
        System.out.println("Imprimindo contrato: " + detalhes);
    }
}

public class Main {
    public static void main(String[] args) {
        Imprimivel relatorio = new Relatorio("Relatório financeiro");
        Imprimivel contrato = new Contrato("Contrato de prestação de serviços");

        relatorio.imprimir();
        contrato.imprimir();
    }
}
